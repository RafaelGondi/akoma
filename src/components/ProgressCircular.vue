<template>
	<svg
		class="progress-circular"
		viewBox="0 0 36 36"
		:height="size"
		:width="size"
	>
		<path
			class="progress-circular__ring-background"
			:stroke="backgroundColor"
			:stroke-width="stroke"
			d="M18 2.0845
			a 15.9155 15.9155 0 0 1 0 31.831
			a 15.9155 15.9155 0 0 1 0 -31.831"
		/>
		<path
			class="progress-circular__ring-progress"
			:stroke="color"
			:stroke-dasharray="normalizedDashArray"
			:stroke-width="stroke-1"
			d="M18 2.0845
			a 15.9155 15.9155 0 0 1 0 31.831
			a 15.9155 15.9155 0 0 1 0 -31.831"
		/>
		<text x="18" y="21.7" :fill="labelColor" class="progress-circular__label">{{label}}</text>
	</svg>
</template>

<script>
export default {
	props: {
		/**
		 * Define a cor da ProgressCircular.
		 */
		color: {
			type: String,
			default: '#2EB88D',
			required: false,
		},
		/**
		 * Define a cor de background da ProgressCircular.
		 */
		backgroundColor: {
			type: String,
			default: '#CED4DA',
			required: false,
		},
		/**
		 * Define o tamanho (diâmetro) da ProgressCircular.
		 */
		size: {
			type: Number,
			default: 24,
			required: false,
		},
		/**
		 * Define o valor correspondente ao percentual ou fração preenchida da ProgressCircular.
		 */
		value: {
			type: Number,
			default: 0,
			required: true,
		},
		/**
		 * Define o valor máximo, a porção toda do processo, da ProgressCircular.
		 * Por padrão o valor máximo é 100 (padrão porcentagem).
		 */
		maxValue: {
			type: Number,
			default: 100,
			required: false,
		},
		/**
		 * Define a espessura do circulo da ProgressCircular.
		 */
		stroke: {
			type: Number,
			default: 2.6,
			required: false,
		},
		/**
		 * Define a cor do texto (label) dentro da ProgressCircular.
		 */
		labelColor: {
			type: String,
			default: '#303A44',
			required: false,
		},
	},

	computed: {
		label() {
			if (this.maxValue >= 100) {
				const normalizedValue = Math.round(100 * this.value / this.maxValue);
				return `${normalizedValue}%`;
			}

			return `${this.value}/${this.maxValue}`;
		},

		normalizedDashArray() {
			const normalizedValue = 100 * this.value / this.maxValue;
			return `${normalizedValue}, 100`;
		},
	},
}
</script>

<style lang="scss" scoped>
.progress-circular {
	display: block;
	margin: 0 auto;
	max-width: 80%;
	max-height: 250px;

	&__ring-background {
		fill: none;
	}

	&__ring-progress {
		fill: none;
		stroke-linecap: round;
		animation: progress 1s ease-out forwards;
	}

	@keyframes progress {
		0% {
			stroke-dasharray: 0 100;
		}
	}

	&__label {
		font-family: sans-serif;
		font-size: 0.7em;
		text-anchor: middle;
	}
}
</style>
