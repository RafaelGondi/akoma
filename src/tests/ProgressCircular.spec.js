import { mount, createLocalVue } from "@vue/test-utils";
import ProgressCircular from "../../src/components/ProgressCircular.vue";

const localVue = createLocalVue();

test("Component is mounted properly", () => {
  const wrapper = mount(ProgressCircular, {
    localVue,
    propsData: {
      value: 33,
    },
  });

  expect(wrapper).toMatchSnapshot();
});
