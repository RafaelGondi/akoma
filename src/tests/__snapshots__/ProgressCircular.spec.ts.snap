// Vitest Snapshot v1

exports[`ProgressCircular > renders correctly 1`] = `
VueWrapper {
  "__app": {
    "_component": [Object],
    "_container": <div … />,
    "_context": [Object],
    "_instance": [Object],
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": [Object],
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": {
    "circleIndicatorStyle": [Object],
    "colorHexCode": [Function],
    "hasOwnProperty": [Function],
    "label": "1%",
    "labelClass": "progress-circular__label--reduced",
    "maxValue": 100,
    "noLabel": false,
    "normalizedDashArray": "0.67, 100",
    "size": "md",
    "slim": false,
    "stroke": 3.5,
    "thick": false,
    "value": 0.67,
    "variant": "green",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <svg
    class="progress-circular progress-circular--md"
    viewBox="0 0 36 36"
  >
    <path … />
    <path … />
    <text … />
  </svg>,
}
`;
